### ISSUE TITLE
[query] Raise a runtime error if projected variables aren't mentioned in the query

### ISSUE THREAD

**Issue by rnewman** [mozilla/mentat#408]
Right now we panic neatly in the projector:

```
                let columns = query.cc
                                   .column_bindings
                                   .get(var)
                                   .expect("Every variable has a binding");
```

**Comment by hemangandhi** [comment]
I assume that snippet is from https://github.com/mozilla/mentat/blob/master/query-projector/src/lib.rs#L254?

Do you intend to return an optional or somehow flag it? Or is the hope to pre-process to account for such cases beforehand?

**Comment by rnewman** [comment]
The pattern we already follow for making sure that `:in` variables are present is to check prior to projection:

https://github.com/mozilla/mentat/blob/master/src/query.rs#L132

It should be relatively straightforward to compare the set of projected variables to the set of mentioned variables at a similar point in time, probably [right here](https://github.com/mozilla/mentat/blob/master/query-projector/src/lib.rs#L604), where we have the spec and we're about to build the projector.

That avoids the need for some error handling throughout the projector.

Note that a query like this is valid:

```
[:find ?x :in ?x :where [(< 10 20)]]
```

so we must look at more than the `:where` clauses.

**Comment by rnewman** [comment]
Fixed in #590.
